models/articles.model.js

```diff
@@ -92,10 +92,33 @@ function incVotesById(article_id, inc_votes) {
   });
 }
 
+function deleteComment(comment_id) {
+  return db
+    .query("SELECT * FROM comments WHERE comment_id = $1;", [comment_id])
+    .then((data) => {
+      if (data.rows.length === 0) {
+        return Promise.reject({
+          status: 404,
+          message: "No such comment_id exist yet",
+        });
+      }
```

Comment:
We can also try do the deletion, and if the delete query returns 0 rows assume that there was no comment with that id. That optional improvement will allow us to do this in one query

Reply:
Got it! I could have explored what would be returned if there was nothing to delete.